### PR TITLE
Print better promote message when unable to find a specification, fix #8064

### LINKED ISSUE #8064: "Unable to find a specification for" error resolution
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run `pod install` on a Podfile specifying a pod that is available upstream but not cached in my local specs repo.

## What did you expect to happen?

An error message with a suggestion to run `pod repo update` for new pods.

## What happened instead?

An error with no suggested resolution:

```
Analyzing dependencies
[!] Unable to find a specification for `PointFree-Validated`
```

### Stack

```
   CocoaPods : 1.6.0.beta.1
        Ruby : ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin18]
    RubyGems : 2.7.6
        Host : Mac OS X 10.14 (18A377a)
       Xcode : 10.0 (10L232m)
         Git : git version 2.17.1 (Apple Git-112)
Ruby lib dir : /Users/stephencelis/.rbenv/versions/2.5.1/lib
Repositories : googlesource-nearby - https://cpdc-eap.googlesource.com/Nearby.git @ eded63d2a99c571e1d01b0aa1b860b99496c3495
               googlesource-spec - https://cpdc-eap.googlesource.com/spec.git @ 0fd9263ede63bee9aec1034b38771950cd7e147a
               master - https://github.com/CocoaPods/Specs.git @ ad98367c1611113ad457fe12f8690359354931a9
```

### Installation Source

```
Executable Path: /Users/stephencelis/.rbenv/versions/2.5.1/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'

target 'MyApp' do
  pod 'PointFree-Validated', '~> 0.1'
end
```

## Project that demonstrates the issue

N/A

### PR DESCRIPTION
**Pull request by X140Yu**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   due to the lack of a generated executable.  
   [Samuel Giddins](https://github.com/segiddins)
 
+* Print better promote message when unable to find a specification.
+  [Xinyu Zhao](https://github.com/X140Yu)
+  [#8071](https://github.com/CocoaPods/CocoaPods/pull/8071)
```
Comment (amorde) [inline CHANGELOG.md]: Can you change this to link to the original issue?

```
[#8064](https://github.com/CocoaPods/CocoaPods/issues/8064)
```

File: lib/cocoapods/resolver.rb [hunk 2]
```diff
@@ -509,6 +509,17 @@ def handle_resolver_error(error)
...
           end,
         )
+      when Molinillo::NoSuchDependencyError
+        message += <<-EOS
+
+
+You have either:
+ * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
+ * mistyped the name or version.
+ * not added the source repo that hosts the Podspec to your Podfile.
```
Comment (amorde) [inline lib/cocoapods/resolver.rb]: this is great!

File: CHANGELOG.md (hunk 1 above)
Comment (X140Yu) [reply CHANGELOG.md]: I've changed it 👌

File: CHANGELOG.md [hunk 3]
```diff
@@ -17,6 +17,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   due to the lack of a generated executable.  
   [Samuel Giddins](https://github.com/segiddins)
 
+* Print better promote message when unable to find a specification.
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit: please add two missing spaces after the period for formatting.

**Comment by X140Yu** [comment]
@amorde Done.

File: CHANGELOG.md (hunk 3 above)
Comment (X140Yu) [reply CHANGELOG.md]: Done.

**Pull request merged**